src/app/core/notifications/notification.model.js

```diff
@@ -19,8 +24,6 @@ const NotificationSchema = new mongoose.Schema(
 		},
 		created: {
 			type: Date,
-			default: () => Date.now(),
-			get: util.dateParse,
 			expires: config.notificationExpires
```

Comment:
While testing to make sure `expires` still works when using `timestamps` I came across a potential issue. If the expires value is changed in the config that will mess up the index the next time the server is run:
```
[2022-09-13T14:17:17.298Z]  INFO: application/48682 on ecapito: Mongoose: Setting debug to false
[2022-09-13T14:17:17.641Z]  INFO: application/48682 on ecapito: Starting initialization of Node.js server
[2022-09-13T14:17:17.703Z]  INFO: application/48682 on ecapito: Mongoose: Connected to "admin" default db
[2022-09-13T14:17:17.936Z] FATAL: application/48682 on ecapito: Mongoose: Could not connect to admin db
[2022-09-13T14:17:17.997Z] FATAL: application/48682 on ecapito: Startup initialization failed. (err.code=85)
    MongoServerError: Index with name: created_1 already exists with different options
        at Connection.onMessage (/Users/ecapito/workspace/node-rest-starter/node_modules/mongoose/node_modules/mongodb/src/cmap/connection.ts:438:20)
        at MessageStream.<anonymous> (/Users/ecapito/workspace/node-rest-starter/node_modules/mongoose/node_modules/mongodb/src/cmap/connection.ts:256:56)
        at MessageStream.emit (node:events:513:28)
        at processIncomingData (/Users/ecapito/workspace/node-rest-starter/node_modules/mongoose/node_modules/mongodb/src/cmap/message_stream.ts:193:14)
        at MessageStream._write (/Users/ecapito/workspace/node-rest-starter/node_modules/mongoose/node_modules/mongodb/src/cmap/message_stream.ts:70:5)
        at writeOrBuffer (node:internal/streams/writable:391:12)
        at _write (node:internal/streams/writable:332:10)
        at MessageStream.Writable.write (node:internal/streams/writable:336:10)
        at Socket.ondata (node:internal/streams/readable:754:22)
        at Socket.emit (node:events:513:28)
```

I changed the expires to `1m` to make sure things were still aging off. This broke and the server would not start until I ran `db.notifications.drop()`. I did verify `expires` still works with timestamps tho 🤷‍♂️

Reply:
This is known behavior with mongoose.  You could also have fixed by just dropping the changed index using `db.collection.dropIndex(indexName)`.  Mongoose also has a [syncIndex](https://thecodebarbarian.com/whats-new-in-mongoose-5-2-syncindexes) function that could be used to resolve this.